lib/rpmsg/rpmsg_virtio.c

```diff
@@ -578,13 +553,6 @@ int rpmsg_init_vdev(struct rpmsg_virtio_device *rvdev,
 #endif /*!VIRTIO_MASTER_ONLY*/
 	rvdev->shbuf_io = shm_io;
 
-#ifndef VIRTIO_MASTER_ONLY
-	if (role == RPMSG_REMOTE) {
-		/* wait synchro with the master */
-		rpmsg_virtio_wait_remote_ready(rvdev);
-	}
-#endif /*!VIRTIO_MASTER_ONLY*/
-
```

Comment:
this is right if remoteproc_create_virtio is used but it is not the case in zephyr sample that uses only rpmsg without the remoteproc_virtio:
https://github.com/zephyrproject-rtos/zephyr/pull/11355/commits/fa8147a265b0317fa070f202bce0b4cc107465d2 
@galak: Could you confirm?

Reply:
Agree with @arnop2, remoteproc_create_virtio() is only used when remoteproc is used. However, RPMsg and virtio is allowed to be used without remoteproc.
We had discussion before on waiting remote to be ready https://github.com/OpenAMP/open-amp/pull/141
We'd better keep checking if virtio ready inside the library, otherwise, additional step will be introduced for user to make sure the virtio is ready before calling the API.

Reply:
Ok, I will remove this patch in the next update